04_documentation/examples/sphinx/src/solver/diffusionsolver.py

```diff
@@ -0,0 +1,175 @@
+"""
+Example acquired from https://scipython.com/book/chapter-7-matplotlib/examples/the-two-dimensional-diffusion-equation/
+"""
+
+import numpy as np
+import matplotlib.pyplot as plt
+
+
+class SolveDiffusion2D:
+    """2D diffusion equation solved with finite differences
+    
+       Methods
+       -------
+       initialize_domain:
+          Initializes domain and mesh of the domain
+       initialize_physical_parameters:
+          Initializes physical parameters and computes time step size
+    """
+
+
+    def __init__(self):
+        """Constructs an uninitialized 2D diffusion solver
+        
+        After construction, initialize domain with `initialize_domain` and 
+        initialize the physical parameters with `initialize_physical_parameters`.
+        """ 
+        # plate size, mm
+        self.w = None
+        self.h = None
+
+        # intervals in x-, y- directions, mm
+        self.dx = None
+        self.dy = None
+
+        # Number of discrete mesh points in X and Y directions
+        self.nx = None
+        self.ny = None
+
+        # Thermal diffusivity of steel, mm^2/s
+        self.D = None
+
+        # Initial cold temperature of square domain
+        self.T_cold = None
+
+        # Initial hot temperature of circular disc at the center
+        self.T_hot = None
+
+        # Timestep
+        self.dt = None
+
+    def initialize_domain(self, w=10., h=10., dx=0.1, dy=0.1):
+        """Initializes domain and mesh of the domain
+        
+        - Sets width and height of the domain. 
+        - Also sets mesh width in x and in y direction.
+        - Using these four values, the method computes the number of mesh elements in x and in y direction.
+        
+        Parameters
+        ----------
+        w : float, default: 10.0
+            width of the domain
+        h : float, default: 10.0
+            height of the domain
+        dx : float, default: 0.1
+            mesh width in x direction
+        dy : float, default: 0.1
+            mesh width in y direction
+        """
+        self.w = w
+        self.h = h
+        self.dx = dx
+        self.dy = dy
+        self.nx = int(w / dx)
+        self.ny = int(h / dy)
+
+    def initialize_physical_parameters(self, d=4., T_cold=300, T_hot=700):
```

Comment:
Is it intended that every function has a complete docstring? If yes, then *Parameters* are missing here. Also some things are missing in functions further down.

Reply:
No, was not my intention :grin: